src/main/java/com/synopsys/integration/alert/web/actions/DescriptorMetadataActions.java

```diff
@@ -0,0 +1,147 @@
+
+    @Autowired
+    public DescriptorMetadataActions(Collection<Descriptor> descriptors, AuthorizationManager authorizationManager) {
+        this.descriptors = descriptors;
+        this.authorizationManager = authorizationManager;
+    }
+
+    public Set<DescriptorMetadata> getDescriptors(@Nullable String name, @Nullable String type, @Nullable String context) {
+        return getDescriptors(name, type, context, this::generateUIComponents);
+    }
+
+    public Set<DescriptorMetadata> getDescriptorsByPermissions(@Nullable String name, @Nullable String type, @Nullable String context) {
+        return getDescriptors(name, type, context, this::generateUIComponentsByPermissions);
+    }
+
+    private Set<DescriptorMetadata> getDescriptors(@Nullable String name, @Nullable String type, @Nullable String context,
+        BiFunction<Set<Descriptor>, ConfigContextEnum, Set<DescriptorMetadata>> generatorFunction) {
+        Predicate<Descriptor> filter = Descriptor::hasUIConfigs;
+        if (name != null) {
+            filter = filter.and(descriptor -> name.equalsIgnoreCase(descriptor.getDescriptorKey().getUniversalKey()));
+        }
+
+        DescriptorType typeEnum = EnumUtils.getEnumIgnoreCase(DescriptorType.class, type);
+        if (typeEnum != null) {
+            filter = filter.and(descriptor -> typeEnum.equals(descriptor.getType()));
+        } else if (type != null) {
+            return Set.of();
+        }
+
+        ConfigContextEnum contextEnum = EnumUtils.getEnumIgnoreCase(ConfigContextEnum.class, context);
+        if (contextEnum != null) {
+            filter = filter.and(descriptor -> descriptor.hasUIConfigForType(contextEnum));
+        } else if (context != null) {
+            return Set.of();
+        }
+
+        Set<Descriptor> filteredDescriptors = filter(descriptors, filter);
+        return generatorFunction.apply(filteredDescriptors, contextEnum);
+    }
+
+    private Set<Descriptor> filter(Collection<Descriptor> descriptors, Predicate<Descriptor> predicate) {
+        return descriptors
+                   .stream()
+                   .filter(predicate)
+                   .collect(Collectors.toSet());
+    }
+
+    private Set<DescriptorMetadata> generateUIComponents(Set<Descriptor> filteredDescriptors, ConfigContextEnum context) {
+        ConfigContextEnum[] applicableContexts = (null != context) ? new ConfigContextEnum[] { context } : ConfigContextEnum.values();
+        Set<DescriptorMetadata> descriptorMetadata = new HashSet<>();
+        for (ConfigContextEnum applicableContext : applicableContexts) {
+            for (Descriptor descriptor : filteredDescriptors) {
+                descriptor.createMetaData(applicableContext)
+                    .ifPresent(descriptorMetadata::add);
+            }
+        }
+        return descriptorMetadata;
+    }
+
+    private Set<DescriptorMetadata> generateUIComponentsByPermissions(Set<Descriptor> filteredDescriptors, ConfigContextEnum context) {
```

Comment:
Create the DescriptorMetadata based on the user permissions.  Refactored the code from the controller because we were making an extra method call that we didn't need.  Check the permissions here.  If the user has permissions then create the DescriptorMetadata objects.